src/record_syscall.cc

```diff
@@ -1745,6 +1745,11 @@ static Switchable prepare_ioctl(RecordTask* t,
       syscall_state.reg_parameter<typename Arch::pid_t>(3);
       return PREVENT_SWITCH;
 
+    case TIOCGSERIAL:
+    case TIOCSSERIAL:
```

Comment:
I think `TIOCSSERIAL` does not need the out-parameter and should just return `PREVENT_SWITCH` right?

Reply:
oh, sure. Sorry about that. I've had those changes sitting unstaged for ages; I'd assumed I had done due diligence when writing them, but I'm pretty sure I'm looking at the documentation for `reg_parameter` for the first time now :P

Out of curiosity, why doesn't `TIOCSSERIAL` need to call `reg_parameter` with `mode = IN`? Is that mostly designed for use with an `ArgMutator`?

Reply:
> Out of curiosity, why doesn't TIOCSSERIAL need to call reg_parameter with mode = IN? Is that mostly designed for use with an ArgMutator?

Yes, more or less. See the comments for ArgMode's IN variant.